plugins/modules/oidc_config.py

```diff
@@ -0,0 +1,153 @@
+      - Write only.
+    type: str
+  certificate:
+    description:
+      - Plain text of the X.509 PEM-encode certificate.
+    type: str
+  config_url:
+    description:
+      - The openID connect provider configuration information endpoint.
+    type: str
+    required: true
+  scopes:
+    description:
+      - Scopes required to obtain necessary claims.
+    type: str
+    required: true
+notes:
+  - C(check_mode) is not supported.
+"""
+
+EXAMPLES = r"""
+- name: New OIDC config
+  scale_computing.hypercore.oidc_config:
+    client_id: 12345
+    shared_secret: secret_stuff
+    certificate: plain_text_from_x509
+    config_url: https:somwhere.com/this/endpoint
+    scopes: required_scopes
+    state: present
+"""
+
+RETURN = r"""
+record:
+  description:
+    - OIDC config record.
+  returned: success
+  type: dict
+  sample:
+    client_id: 1234
+    certificate: this_certificate
+    config_url: https://somewhere.com/this/endpoint
+    scopes: required_scopes
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils import arguments, errors
+from ..module_utils.utils import is_changed
+from ..module_utils.client import Client
+from ..module_utils.rest_client import RestClient
+from ..module_utils.oidc import Oidc
+from ..module_utils.typed_classes import TypedOidcToAnsible, TypedDiff
+from ..module_utils.task_tag import TaskTag
+from typing import Union, Tuple
+from time import sleep
+
+
+def ensure_present(
+    module: AnsibleModule,
+    rest_client: RestClient,
+    oidc_obj: Union[Oidc, None],
+    unit: bool,
+) -> Tuple[bool, Union[TypedOidcToAnsible, None], TypedDiff]:
+    before = oidc_obj.to_ansible() if oidc_obj else None
+    oidc_obj_ansible = Oidc.from_ansible(module.params)
+    if oidc_obj is None:
+        task = oidc_obj_ansible.send_create_request(rest_client)
+    else:
+        task = oidc_obj_ansible.send_update_request(rest_client)
+    if not unit:
+        sleep(10)  # Wait for the cluster login (Avoid BAD GATEWAY response)
+    TaskTag.wait_task(rest_client, task)
+    updated_oidc = Oidc.get(rest_client)
+    after = updated_oidc.to_ansible() if updated_oidc else None
+    return is_changed(before, after), after, dict(before=before, after=after)
+
+
+def run(
+    module: AnsibleModule, rest_client: RestClient, unit: bool = False
+) -> Tuple[bool, Union[TypedOidcToAnsible, None], TypedDiff]:
+    oidc_obj = Oidc.get(rest_client)
+    return ensure_present(module, rest_client, oidc_obj, unit)
+
+
+def main() -> None:
+    module = AnsibleModule(
+        supports_check_mode=False,
+        argument_spec=dict(
+            arguments.get_spec("cluster_instance"),
+            client_id=dict(
+                type="str",
+                required=True,
+            ),
+            shared_secret=dict(
+                type="str",
+                no_log=False,
+            ),
+            certificate=dict(
+                type="str",
+            ),
```

Comment:
`certificate` is used for client certificate authentication? If yes, please add `no_log: True`.

Reply:
Yes I think it is. You can use it instead of the shared_secret. Nice catch.